### ISSUE TITLE
google_storage_bucket_iam_policy.my_policy infinite updated loop on terraform apply

### ISSUE THREAD

**Issue by vdamianchicon** [terraform-providers/terraform-provider-google#2501]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment
* If an issue is assigned to the "modular-magician" user, it is either in the process of being autogenerated, or is planned to be autogenerated soon. If an issue is assigned to a user, that user is claiming responsibility for the issue. If an issue is assigned to "hashibot", a community member has claimed the issue already.

### Terraform Version
Terraform v0.11.10
+ provider.google v1.19.1
+ provider.google-beta v1.19.0

### Affected Resource(s)

* google_storage_bucket_iam_policy

### Terraform Configuration Files

```tf
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://www.hashicorp.com/security
# If reproducing the bug involves modifying the config file (e.g., apply a config,
# change a value, apply the config again, see the bug) then please include both the
# version of the config before the change, and the version of the config after the change.
```

### Expected Behavior
```
resource "google_storage_bucket" "my_bucket" {
  name     = "my_bucket"
  versioning = {
    enabled = "true"
  }
}

data "google_iam_policy" "my_bucket_policy_bindings" {
  binding {
    role = "roles/storage.legacyBucketOwner"
    members = "${local.my_bucket_legacyBucketOwner}"
  }
  binding {
    role = "roles/storage.legacyBucketReader"
    members = "${local.my_bucket_legacyBucketReader}"
  }
  binding {
    role = "roles/storage.admin"
    members = "${local.my_bucket_admin}"
  }
  binding {
    role = "roles/storage.objectViewer"
    members = "${local.my_bucket_objectViewer}"
  }
}

resource "google_storage_bucket_iam_policy" "my_bucket_policy" {
  bucket = "my_bucket"
  policy_data = "${data.google_iam_policy.my_bucket_policy_bindings.policy_data}"
}

resource "google_storage_bucket" "my_other_bucket" {
  name     = "my_other_bucket"
  versioning = {
    enabled = "true"
  }
}

data "google_iam_policy" "my_other_bucket_policy_bindings" {
  binding {
    role = "roles/storage.legacyBucketOwner"
    members = "${local.v_legacyBucketOwner}"
  }
  binding {
    role = "roles/storage.legacyBucketReader"
    members = "${local.my_other_bucket_legacyBucketReader}"
  }
  binding {
    role = "roles/storage.admin"
    members = "${local.my_other_bucket_admin}"
  }
  binding {
    role = "roles/storage.objectViewer"
    members = "${local.my_other_bucket_objectViewer}"
  }
}

resource "google_storage_bucket_iam_policy" "my_other_bucket_policy" {
  bucket = "my_other_bucket"
  policy_data = "${data.google_iam_policy.my_other_bucket_policy_bindings.policy_data}"
}


Terraform will perform the following actions:

    ~ google_storage_bucket_iam_policy.my_bucket_policy
       20-Nov-2018 13:57:22	      policy_data: 
       ...from my_bucket_policy_bindings => 
       ...to my_other_bucket_policy

    ~ google_storage_bucket_iam_policy.my_other_bucket_policy
       20-Nov-2018 13:57:22	      policy_data: 
       ...from my_other_bucket_policy => 
       ...to my_bucket_policy_bindings

Plan: 0 to add, 2 to change, 0 to destroy.

```

### Actual Behavior
```
NOTE: on every plan/apply the google_storage_bucket_iam_policy changes
repeated for ever


Terraform will perform the following actions:

    ~ google_storage_bucket_iam_policy.my_bucket_policy
       20-Nov-2018 13:57:22	      policy_data: 
       ...from my_bucket_policy_bindings => 
       ...to my_other_bucket_policy

    ~ google_storage_bucket_iam_policy.my_other_bucket_policy
       20-Nov-2018 13:57:22	      policy_data: 
       ...from my_other_bucket_policy => 
       ...to my_bucket_policy_bindings

Plan: 0 to add, 2 to change, 0 to destroy.

```

### Steps to Reproduce

1. `terraform apply`

### Important Factoids

It seems that we can only have one "IAM access control policy" (aka terraform's google_iam_policy) per project.
because terraform's google_iam_policy uses google cloud api Method: projects.setIamPolicy which 
"Sets the IAM access control policy for the specified Project. Overwrites any existing policy."
In our use case looks like 
data "google_iam_policy" "old_policy_bindings" was being overwritten by
data "google_iam_policy" "new_policy_bindings" and vice versa in each subsequent terraform apply
unless I'm mistaking, it causes
google_storage_bucket_iam_policy.old_policy
and google_storage_bucket_iam_policy.new_policy to be in an infinite updating loop.

Could you take a look please

### References

* #0000

**Comment by vdamianchicon** [comment]
Paddy thank you for the quick response. I have actually redone the code step by step and it works now. 
I was trying to reproduced the problem to send you logs. However I was not able to reproduced it. I tried 2 time in a test environment and once in production. So, I think this issue can be closed.
As to what happened when I experienced this issue, 3 things to consider:
1.  I got a few state lock errors and had to unlock and even delete the lock file manually.
2. I may have place variables in the wrong place causing the raise condition.
3. Before I started trying to reproduced the issue, There was a SDK upgrade from google cloud which I deployed.
Your current Cloud SDK version is: 225.0.0
You will be upgraded to version: 226.0.0

I think any of the 3 factors may have contribute to my issue.
Thanks and sorry for the false alarm : )
